content/learning-paths/cross-platform/simd-info-demo/simdinfo-example1-porting.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Code Verification
+weight: 6
+
+### FIXED, DO NOT MODIFY
+layout: learningpathall
+---
+
+### Step-by-Step Porting
+
+1. Change the loading process to follow **NEON**'s method for initializing vectors. The **SSE4.2** intrinsic **`_mm_set_ps`** is in reality a macro, in **NEON** you can do the same thing with curly braces **`{}`** inititialization.
+2. Next, you will replace the **SSE4.2** intrinsics with the **NEON** equivalents we identified earlier. The key is to ensure that the operations perform the same tasks, such as comparison, addition, multiplication, and square root calculations.
+3. Finally, modify the storing process to match **NEON**’s way of moving data from vectors to memory. In **NEON**, you use functions like [**`vst1q_f32`**](https://simd.info/c_intrinsic/vst1q_f32/) for storing 128-bit floating-point vectors and [**`vst1q_u32`**](https://simd.info/c_intrinsic/vst1q_u32/) for storing 128-bit integer vectors.
+
+After identifying the **NEON** intrinsics you will need in the ported program, it's time to actually write the code.
+
+Create a new file for the ported NEON code named `calculation_neon.c` with the contents shown below:
+
+```C
+#include <arm_neon.h>
+#include <stdio.h>
+
+int main() {
+    float32x4_t a = {1.0f, 4.0f, 9.0f, 16.0f};
+    float32x4_t b = {1.0f, 2.0f, 3.0f, 4.0f};
+
+    uint32x4_t cmp_result = vcgtq_f32(a, b);
+
+    float a_arr[4], b_arr[4];
+    uint32_t cmp_res[4];
+
+    vst1q_f32(a_arr, a);
+    vst1q_f32(b_arr, b);
+    vst1q_u32(cmp_res, cmp_result);
+
+    for (int i = 0; i < 4; i++) {
+        if (cmp_res[i] != 0) {
+            printf("Element %d: %.2f is larger than %.2f\n", i, a_arr[i], b_arr[i]);
+        } else {
+            printf("Element %d: %.2f is not larger than %.2f\n", i, a_arr[i], b_arr[i]);
+        }
+    }
+    printf("\n");
+
+    float32x4_t add_result = vaddq_f32(a, b);
+    float32x4_t mul_result = vmulq_f32(add_result, b);
+    float32x4_t sqrt_result = vsqrtq_f32(mul_result);
+
+    float res[4];
+
+    vst1q_f32(res, add_result);
+    printf("Addition Result: %.2f %.2f %.2f %.2f\n", res[0], res[1], res[2], res[3]);
+
+    vst1q_f32(res, mul_result);
+    printf("Multiplication Result: %.2f %.2f %.2f %.2f\n", res[0], res[1], res[2], res[3]);
+
+    vst1q_f32(res, sqrt_result);
+    printf("Square Root Result: %.2f %.2f %.2f %.2f\n", res[0], res[1], res[2], res[3]);
+
+    return 0;
+}
+```
+
+### Verifying the Ported Code
+
+It's time to verify that the functionality remains the same, which means you get the same results and similar performance.
+
+Compile the above code as follows on an Arm system:
+
+```bash
+gcc -O3 calculation_neon.c -o calculation_neon
```

Comment:
Because the constants are local, the compiler constant evaluates this example entirely at compile time https://godbolt.org/z/eeY754815 so there won't be any Adv. SIMD instructions generated here (aside from `fsqrt`).  It may be confusing for someone verifying the source.

You could consider lifting the constants to global scope to prevent this: https://godbolt.org/z/4qjxPd399

Same for the other examples

Reply:
Thanks for the comment, we changed the examples to use global scope.